DQMOffline/Trigger/src/HLTMuonMatchAndPlot.cc

```diff
@@ -374,37 +376,35 @@ vector<size_t> HLTMuonMatchAndPlot::matchByDeltaR(const vector<T1>& collection1,
 
 MuonCollection HLTMuonMatchAndPlot::selectedMuons(const MuonCollection& allMuons,
                                                   const BeamSpot& beamSpot,
-                                                  bool hasRecoCuts,
-                                                  const StringCutObjectSelector<reco::Muon>& selector,
+                                                  double RecoMuonEtaMax,
+                                                  double RecoMuonEtaMin,
+                                                  bool IsMuonGlb,
                                                   double d0Cut,
                                                   double z0Cut) {
-  // If there is no selector (recoCuts does not exists), return an empty collection.
-  if (!hasRecoCuts)
-    return MuonCollection();
-
   MuonCollection reducedMuons;
   for (auto const& mu : allMuons) {
     const Track* track = nullptr;
     if (mu.isTrackerMuon())
       track = &*mu.innerTrack();
     else if (mu.isStandAloneMuon())
       track = &*mu.outerTrack();
-    if (track && selector(mu) && fabs(track->dxy(beamSpot.position())) < d0Cut &&
-        fabs(track->dz(beamSpot.position())) < z0Cut)
+    bool muID = true;
+    if (IsMuonGlb)
+      muID = mu.isGlobalMuon();
+    else
+      muID = mu.isStandAloneMuon();  // minimun ID (requested for cosmics)
+    if (track && muID && abs(mu.eta()) < RecoMuonEtaMax && abs(mu.eta()) >= RecoMuonEtaMin &&
+        fabs(track->dxy(beamSpot.position())) < d0Cut && fabs(track->dz(beamSpot.position())) < z0Cut)
       reducedMuons.push_back(mu);
   }
 
   return reducedMuons;
 }
 
-TriggerObjectCollection HLTMuonMatchAndPlot::selectedTriggerObjects(
-    const TriggerObjectCollection& triggerObjects,
-    const TriggerEvent& triggerSummary,
-    bool hasTriggerCuts,
-    const StringCutObjectSelector<TriggerObject>& triggerSelector) {
-  if (!hasTriggerCuts)
-    return TriggerObjectCollection();
-
+TriggerObjectCollection HLTMuonMatchAndPlot::selectedTriggerObjects(const TriggerObjectCollection& triggerObjects,
+                                                                    const TriggerEvent& triggerSummary,
+                                                                    double triggerEtaMaxCut,
+                                                                    double triggerEtaMinCut) {
```

Comment:
These are all class members: I don't see the reason why to pass them as parameters to the method (it does not seem to me that they get modified by the method)
```suggestion
                                                                    ) {
```